### PR TITLE
Document good and possibly poor use-cases of Bitcask

### LINKED ISSUE #193: Evict inactive keys from memory to disk to conserve memory
As your database grows, some proportion of keys can potentially become infrequently used or inactive, depending on your use-case and access patterns.

Instead of holding the entire key space in memory at all times, what if we evict infrequently or inactive keys to disk?

### PR DESCRIPTION
**Pull request by prologic**
Fixes #193

### CODE REVIEW

**Comment by prologic** [comment]
@yashschandra @Hassall What do you think?

File: README.md [hunk 1]
```diff
@@ -19,13 +19,67 @@ For a more feature-complete Redis-compatible server, distributed key/value store
...
+  or [LevelDB](https://github.com/syndtr/goleveldb) if you intend to store a
+  large quantity of key/value pairs.
+
+> Note however that storing large amounts of data in terms of value(s) is
+> totally fine. In other wise thousands to millions of keys with large values
+> will work just fine.
+
+- Write intensive workloads. Due to the [Bitcask design](https://riak.com/assets/bitcask-intro.pdf?source=post_page---------------------------)
+  heavy write workloads that lots of key/value pairs will over time cause
+  problems like "Too many open files" errors to occur. This can be mitigated by
```
Comment (yashschandra) [inline README.md:74]: should we link the issue also with `too many open files` ?

Comment (prologic) [reply README.md:74]: Yea possibly

Comment (prologic) [reply README.md]: Done

**Comment by yashschandra** [comment]
looks good

**Review by yashschandra — APPROVED** [review]

**Pull request merged**